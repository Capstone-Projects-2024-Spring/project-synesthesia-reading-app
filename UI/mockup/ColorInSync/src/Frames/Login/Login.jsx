import { useState } from "react";
import "./Login.css";
import { GoogleLogin, googleLogout, useGoogleLogin } from "@react-oauth/google";

function Login({ loginSuccess }) {
  const login = useGoogleLogin({
    onSuccess: (tokenResponse) => loginSuccess(tokenResponse),
  });
  const errorMessage = (error) => {
    console.log(error);
  };
  const logOut = () => {
    googleLogout();
    setProfile(null);
  };
  return (
    <>
      <div className="bg-gray-200 flex flex-col items-center space-y-10 justify-center h-screen">
        <h1 className="text-3xl font-bold text-purple-500">Color in Sync</h1>
        <button className="bg-gray-100" onClick={() => login()}>
          Log in with Google
        </button>

        <button className="bg-purple-500 text-gray-200">
          Continue as guest
        </button>
      </div>
    </>
  );
}

export default Login;
