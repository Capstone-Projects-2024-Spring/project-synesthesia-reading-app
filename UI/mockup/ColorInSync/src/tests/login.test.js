import renderer from 'react-test-renderer'
import Login from './../Frames//Login/Login.jsx'
import { GoogleOAuthProvider } from "@react-oauth/google";

it('renders correctly', () => {
    const tree = renderer.create( 
        <GoogleOAuthProvider clientId={process.env.VITE_GOOGLE_OATH_KEY}>
            <Login loginSuccess = {(user) => console.log(user)}></Login>
        </GoogleOAuthProvider>)
    .toJSON();
    expect(tree).toMatchSnapshot()

})