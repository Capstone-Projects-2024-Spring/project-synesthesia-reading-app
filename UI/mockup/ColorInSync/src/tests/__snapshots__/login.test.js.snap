// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="bg-gray-200 flex flex-col items-center space-y-10 justify-center h-screen"
>
  <h1
    className="text-3xl font-bold text-purple-500"
  >
    Color in Sync
  </h1>
  <button
    className="bg-gray-100 p-2.5 rounded-md"
    onClick={[Function]}
  >
    Log in with Google
  </button>
  <button
    className="bg-purple-500 text-gray-200 p-3 rounded-md"
  >
    Continue as guest
  </button>
</div>
`;
